ocean_provider/validation/requests.py

```diff
@@ -48,6 +78,150 @@ def __init__(
         self._processor = CustomRulesProcessor()
 
 
+class RBACValidator:
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self._request = request
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._assets = assets
+        self._algorithms = algorithms
+        if self.assets is None:
+            self._assets = []
+        if self.algorithms is None:
+            self._algorithms = []
+        self._compute_dids = {
+            "dids": [
+                {
+                    "did": asset.did,
+                    "serviceId": ServiceTypesIndices.DEFAULT_COMPUTING_INDEX,
+                }
+                for asset in self.assets
+            ],
+            "algos": [
+                {
+                    "did": algorithm.did,
+                    "serviceId": ServiceTypesIndices.DEFAULT_COMPUTING_INDEX,
+                }
+                for algorithm in self.algorithms
+            ],
+        }
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @property
+    def provider_address(self):
+        return get_provider_wallet().address
+
+    @property
+    def component(self):
+        return self._component
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def assets(self):
+        return self._assets
+
+    @property
+    def request(self):
+        return self._request
+
+    @property
+    def compute_dids(self):
+        return self._compute_dids
+
+    @property
+    def algorithms(self):
+        return self._algorithms
+
+    def fails(self):
+        return False
+
+    def msg_hash(self, message: str):
+        return hashlib.sha256(message.encode("utf-8")).hexdigest()
+
+    def build_payload(self):
+        payload = {
+            "eventType": self.action,
+            "component": self.component,
+            "providerAddress": self.provider_address,
+            "credentials": self.credentials,
+        }
+        payload.update(getattr(self, f"build_{self.action}_payload")())
```

Comment:
can you please leave a comment here e.g. "builds actions like build_encrtyptUrl_payload to update the dictionary with request-specific key-values." (or something similar)